File: 7_guessing_game/guessing_game.jl
function get_guess()
    print("Guess a number between 1 and 100: ")
    guess = readline()
    guess = parse(Int, guess)
    return guess
end

secret = 47
remaining_guesses = 10

while remaining_guesses > 0
    global secret
    local guess = get_guess()
    if guess == secret
        print("You win!")
        break
    elseif guess > secret
        println("Lower.")
    else
        println("Higher.")
    end
    global remaining_guesses -= 1
end

if remaining_guesses == 0
    print("Sorry, you lose.")
end

File: 7_guessing_game/guessing_game.cpp
#include <iostream>

using namespace std;

int get_guess() {
    cout << "Guess a number between 1 and 100: ";
    int guess;
    cin >> guess;
    
    return guess;
};

int main() {
    int secret = 47, remaining_guesses = 10;

    while (remaining_guesses > 0) {
        int guess = get_guess();
        if (guess == secret) {
            cout << "You win!";
            break;
        } else if (guess > secret) {
            cout << "Lower." << endl;
        } else {
            cout << "Higher." << endl;
        };
        remaining_guesses -= 1;
    };

    if (remaining_guesses == 0) {
        cout << "Sorry, you lose.";
    };
};